app/src/org/commcare/android/adapters/SquareButtonViewHolder.java

```diff
@@ -0,0 +1,30 @@
+package org.commcare.android.adapters;
+
+import android.support.v7.widget.RecyclerView;
+import android.view.View;
+import android.widget.RelativeLayout;
+import android.widget.TextView;
+
+import org.commcare.android.view.SquareImageView;
+import org.commcare.dalvik.R;
+
+/**
+ * Holds views for a home screen button
+ *
+ * @author Phillip Mates (pmates@dimagi.com).
+ */
+public class SquareButtonViewHolder extends RecyclerView.ViewHolder {
+    public final SquareImageView imageView;
+    public final RelativeLayout cardView;
+    public final TextView textView;
+    public final TextView subTextView;
+
+    public SquareButtonViewHolder(View view) {
```

Comment:
It's kind of an odd pattern to have the constructor take in a View parameter, when you are always explicitly expecting it to be 1 very specific layout file (home_card.xml). Can you just make the constructor take no params and inflate that view within the class? Same goes for the HeaderItemView constructor.